webbpsf/webbpsf_core.py

```diff
@@ -1564,6 +1720,10 @@ def show_notebook_interface(instrument):
     from IPython.display import display, clear_output
     from matplotlib import pyplot as plt
 
+
+    if isinstance(instrument, string_types):
```

Comment:
From the user-typing-things-in-a-notebook user experience perspective, I agree the ergonomics are better. It's a bit of received wisdom that functions should not mix strings containing names for things and the actual things themselves (well, their identifiers in your programming language). It has consequences for testing, leads to multiple concerns entangled in a single function, and the like.

I would rather have a `show_notebook_interface` helper function that does the name lookup / instantiation and calls the "real" `show_notebook_interface` (which could be `webbpsf.nbinterface.construct_interface(inst)` or something). That would keep the concerns separate.